### PR TITLE
[opencascade] Update port to 7.7.2

### PR DESCRIPTION
**Pull request by dfkki123508**
If this PR updates an existing port, please uncomment and fill out this checklist:

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [x] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [x] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dfkki123508** [comment]
@microsoft-github-policy-service agree

File: ports/opencascade/portfile.cmake [hunk 1]
```diff
@@ -23,10 +23,19 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
...
         "tbb"        USE_TBB
         "rapidjson"  USE_RAPIDJSON
-        "samples"    INSTALL_SAMPLES 
+        "samples"    INSTALL_SAMPLES
+        "vtk"        USE_VTK
 )
 
-# VTK option in opencascade not currently supported because only 6.1.0 is supported but vcpkg has >= 9.0
+if(USE_VTK)
+    get_filename_component(VTK_DIR "${CURRENT_PACKAGES_DIR}/../vtk_${TARGET_TRIPLET}" ABSOLUTE)
```
Comment (Neumann-A) [inline ports/opencascade/portfile.cmake:31]: That is wrong on so many levels. Also why do you need to define all those variables? `find_package(VTK REQUIRED CONFIG)` should work fine

File: ports/opencascade/vcpkg.json [hunk 2]
```diff
@@ -39,6 +39,12 @@
       "dependencies": [
         "tbb"
       ]
+    },
+    "vtk": {
+      "description": "Enable optional usage of vtk",
+      "dependencies": [
+        "vtk"
+      ]
```
Comment (Neumann-A) [inline ports/opencascade/vcpkg.json:47]: THis is probably also incorrect. What modules of vtk does opencascade actually require/need ?

File: ports/opencascade/vcpkg.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "opencascade",
-  "version": "7.6.2",
-  "port-version": 7,
+  "version": "7.7.2",
+  "port-version": 8,
```
Comment (JonLiu1993) [inline ports/opencascade/vcpkg.json:4]:
```suggestion
```

Comment (JonLiu1993) [reply ports/opencascade/vcpkg.json:4]: If we update port version, we should remove "port-version".

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by dfkki123508** [comment]
Thanks guys for the review. I updated the files and removed the VTK paths from the portfile, but had to create a separate patch file, since the one from opencascade does not set the include path.. @Neumann-A I am not sure on which vtk modules exactly opencascade depends on. Why can't it remain depending on vtk[core]?

**Comment by Neumann-A** [comment]
> Why can't it remain depending on vtk[core]?

You are depending on `vtk[default]` not `vtk[core]`. I don't mind `vtk[core]` if it is correct.

**Comment by dfkki123508** [comment]
> You are depending on vtk[default] not vtk[core]. I don't mind vtk[core] if it is correct.

My bad, now it should be correct.

**Comment by Neumann-A** [comment]
> My bad, now it should be correct.

no it is not. You need `default-features: "false"`

**Comment by JonLiu1993** [comment]
@dfkki123508, when I tested the features by command `./vcpkg install opencascade[*]:x86-windows`, I get this error:
````
optimized;F:/Feature-test/opencascade/vcpkg/installed/x86-windows/lib/lzma.lib;debug;F:/Feature-test/opencascade/vcpkg/installed/x86-windows/debug/lib/lzma.lib (found version "5.4.3") 
-- Found Threads: TRUE  
VTK version (9.2.20220823)
CMake Error at adm/cmake/tbb.cmake:140 (message):
  Installation directory with TBB is not exist.
Call Stack (most recent call first):
  CMakeLists.txt:34 (include)
  CMakeLists.txt:736 (OCCT_INCLUDE_CMAKE_FILE)


-- Configuring incomplete, errors occurred!
````
Please take a look.

**Comment by dfkki123508** [comment]
Thanks @Neumann-A and @JonLiu1993 . I resolved the issues and built it successfully for x64-windows, x86-windows and x64-windows-static triplets.

**Comment by JonLiu1993** [comment]
@dfkki123508, I tested all features successfully in the follwoing triplet:
- x86-windows
- x64-windows
- x64-windows-static

but when I tested usage by `opencascade[*]:x64-windows` I get this error:
CMakeListe:
````
cmake_minimum_required (VERSION 3.8)

project(test)

find_package(OpenCASCADE CONFIG REQUIRED)
# Add source to this project's executable.
add_executable (test "usageTest.cpp")

target_link_libraries(test PRIVATE TKBO TKBin TKCAF TKCDF)
````
usageTest.cpp:
````
#include <stdio.h>
#include <opencascade/AIS_SignatureFilter.hxx>

int main()
{
    printf("%s\n", "HELLO WORLD");
    return 0;
}
````
Config:
````
1> CMake generation started for configuration: 'x64-Debug'.
1> Command line: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\usageTest\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/opencascade/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\usageTest" 2>&1"
1> Working directory: C:\Users\test\source\repos\usageTest\out\build\x64-Debug
1> [CMake] -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
1> [CMake] -- The C compiler identification is MSVC 19.36.32538.0
1> [CMake] -- The CXX compiler identification is MSVC 19.36.32538.0
1> [CMake] -- Detecting C compiler ABI info
1> [CMake] -- Detecting C compiler ABI info - done
1> [CMake] -- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting C compile features
1> [CMake] -- Detecting C compile features - done
1> [CMake] -- Detecting CXX compiler ABI info
1> [CMake] -- Detecting CXX compiler ABI info - done
1> [CMake] -- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting CXX compile features
1> [CMake] -- Detecting CXX compile features - done
1> [CMake] CMake Warning (dev) at F:/Feature-test/opencascade/vcpkg/installed/x64-windows/share/opencascade/OpenCASCADEConfig.cmake:50:
1> [CMake]   Syntax Warning in cmake code at column 96
1> [CMake] 
1> [CMake]   Argument not separated from preceding token by whitespace.
1> [CMake] Call Stack (most recent call first):
1> [CMake]   F:/Feature-test/opencascade/vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
1> [CMake]   usageTest/CMakeLists.txt:5 (find_package)
1> [CMake] This warning is for project developers.  Use -Wno-dev to suppress it.
1> [CMake] 
1> [CMake] -- Found ZLIB: optimized;F:/Feature-test/opencascade/vcpkg/installed/x64-windows/lib/zlib.lib;debug;F:/Feature-test/opencascade/vcpkg/installed/x64-windows/debug/lib/zlibd.lib (found version "1.3.0")  
1> [CMake] -- Found OpenGL: opengl32   
1> [CMake] -- Configuring done (9.9s)
1> [CMake] -- Generating done (0.1s)
1> [CMake] -- Build files have been written to: C:/Users/test/source/repos/usageTest/out/build/x64-debug
1> Extracted CMake variables.
1> Extracted source files and headers.
1> Extracted code model.
1> Extracted toolchain configurations.
1> Extracted includes paths.
1> CMake generation finished.
````
build:
````
>------ Build All started: Project: usageTest, Configuration: x64-Debug ------
  MSBuild version 17.6.3+07e294721 for .NET Framework
  
    Checking Build System
    Building Custom Rule C:/Users/test/source/repos/usageTest/usageTest/CMakeLists.txt
    usageTest.cpp
C:\Users\test\source\repos\usageTest\out\build\x64-Debug\usageTest\LINK : fatal error LNK1104: cannot open file 'tbb12.lib' 

Build All failed.
````

Please take a look.

**Comment by JonLiu1993** [comment]
Ping @dfkki123508 for response.

**Comment by dfkki123508** [comment]
Hi @JonLiu1993, I updated the tbb-patch, but it's kind of ugly... Your usageTest.cpp works on x64-windows.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
Tested usage successfully by `opencascade:x64-windows`:
````
opencascade provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(OpenCASCADE CONFIG REQUIRED)
    # note: 48 additional targets are not displayed.
    target_link_libraries(main PRIVATE TKBO TKBin TKCAF TKCDF)
````

**Comment by BillyONeal** [comment]
@dfkki123508 Thanks for the update!
@Neumann-A Thanks for the review!
@JonLiu1993 Thanks for testing!

**Pull request merged**